forge-gui/src/main/java/forge/gamemodes/match/input/InputPassPriority.java

```diff
@@ -124,8 +124,8 @@ private void passPriority(final Runnable runnable) {
     @Override
     protected final void onPlayerSelected(Player selected, final ITriggerEvent triggerEvent) {
         PlayerController pc = selected.getController();
-        if (pc.isGuiPlayer()) {
-           pc.setFullControl(!pc.isFullControl());
+        if (!pc.isAI()) {
```

Comment:
Isn't that check broken for network player?

Reply:
What's the reason for hooking the opening and closing of this menu into the input system? Seems like something that should be entirely on the UI side.

Reply:
Hmn but this is still inside forge-gui?
Anywhere higher up the chain seems problematic as we wouldn't know we're in safe state for this (no other input active)

Reply:
What I was getting at is that opening a menu seems like something lower down the chain - handled entirely within the desktop and mobile UI implementations. Right now it appears to be hooked into the input function for choosing a player, which is responding with an instruction to open a menu if a chosen player isn't currently needed.

I guess this works as a way to open the menu when clicking on a player portrait when it wouldn't otherwise do anything. But I would think the input state logic shouldn't be concerned with how the UI makes the Full Control settings available to the user - whether it's a menu opened by clicking a portrait, right clicking in the details panel, a dropdown at the top of the screen, etc. 

You might consider either detecting the input state entirely within the front end (the mana pool icons in [CField.java](https://github.com/Card-Forge/forge/blob/c2015bf0201b3b4f793def0f894cd3c3b03f0f31/forge-gui-desktop/src/main/java/forge/screens/match/controllers/CField.java#L76-L78) do this), or opening the menu through other gestures (right click, long press, swipe, etc.) so as not to share with the "choose player" input.

Reply:
Well I didn't really consider changing the options in the middle of targeting worthwhile.
I could switch to right click and long press instead, though those ``instanceof`` checks in CField aren't looking clean either...
Guess I'll sleep on it :)

Reply:
should be fine now